### PR TITLE
Adds random fortunes to fortune cookies

### PR DESCRIPTION
**Pull request by Citinited**
Just a little immersion addition to the kitchen. This PR does the following:

- If a blank paper is used in the creation of a fortune cookie, it will drop its own nicely-formatted fortune (instead of creating a burnt mess). Now with 100% more prophecy!
- Added height and width variables to paper so that differently-sized pieces of paper are possible.

If anyone has suggestions on good fortunes, let me know. 21 probably isn't enough.

I thought of maybe adding a chance to give useful information in the lucky numbers. Nothing huge like the nuke code (ehehehehe), but again, let me know if you think of anything.

image

image

:cl:
add: Fortune cookies now drop random fortunes if cooked with a blank piece of paper.
/:cl:

### CODE REVIEW

**Comment by Ty-Omaha** [comment]
http://www.fortunecookiemessage.com/archive.php

**Comment by Citinited** [comment]
Yeah, I should probably just copy-paste a whole bunch of those instead of cherry-picking

**Comment by Shadeykins** [comment]
That wasn't a bug insofar as the chef neglected to put a fortune in.

They're designed for the chef to put paper inside as part of the recipe, and the writing on the paper becomes the fortune. IE: I once ate a fortune cookie and my fortune was "Go fuck yourself."

Does this PR preserve that mechanic?

**Comment by FalseIncarnate** [comment]
It appears that this PR does preserve the ability to make your own fortunes... with the unintended effect of adding one of these generic fortunes alongside it.

Considering the fortune cookie recipe should be failing unless there is data on the paper, this is fixing only spawned (not cooked) fortune cookies from not having any sort of fortune.

Additionally, the cooked version sets the fortune as the trash object, not using Post_Consume for something already handled by trash. What you could do is set the default trash item as a fortune paper, so a chef-made version will overwrite the generic fortune in favor of a player-written one.

Here's what I am suggesting you do:
`trash = /obj/item/weapon/paper/fortune` and then delete the Post_Consume and New overrides and the F variable you added.

This preserves the original behavior, ensures that spawned cookies get a fortune, and avoids double fortunes.

File: code/_globalvars/lists/fortunes.dm [hunk 1]
```diff
@@ -0,0 +1,21 @@
+var/global/list/fortune_cookie_messages = list("A woman named [pick(first_names_female)] may have an opportunity for you.",
```
Comment (tigercat2000) [inline code/_globalvars/lists/fortunes.dm]: Use `GLOBAL_LIST_INIT(fortune_cookie_messages, list(...))`

File: code/modules/paperwork/paper.dm [hunk 2]
```diff
@@ -476,6 +478,21 @@
...
 
+/obj/item/weapon/paper/fortune
+	name = "fortune"
+	icon_state = "slip"
+	paper_height = 150
+	var/fortunemessage
+
+/obj/item/weapon/paper/fortune/New()
+	..()
+	var/fortunemessage = pick(fortune_cookie_messages)
```
Comment (tigercat2000) [inline code/modules/paperwork/paper.dm]: And do `pick(GLOB.fortune_cookie_messages)`

**Comment by Citinited** [comment]
@Shadeykins @FalseIncarnate Ah, right, I'll get to preserving that behaviour as it is currently. Didn't realise it worked like that.

Also, the bug comment was merely a jest, though it seems like it might be misleading. I'll update the OP.

**Comment by Citinited** [comment]
@FalseIncarnate @tigercat2000 I have updated this PR with your feedback.

**Comment by taukausanake** [comment]
I got a fortune for you:

"When fighting for freedom, never wear new pants."

A cookie for anyone who can guess where that is from

**Comment by Citinited** [comment]
Added more fortunes, think this is ready for merging and review.

**Comment by Citinited** [comment]
@Fox-McCloud Merge conflicts resolved

**Comment by Fox-McCloud** [comment]
Thank you.

File: code/modules/paperwork/paper.dm [hunk 3]
```diff
@@ -476,6 +478,21 @@
 /obj/item/paper/crumpled/bloody
 	icon_state = "scrap_bloodied"
 
+/obj/item/paper/fortune
+	name = "fortune"
+	icon_state = "slip"
+	paper_height = 150
+	var/fortunemessage
```
Comment (Citinited) [inline code/modules/paperwork/paper.dm]: I need to remove this unused var when I next get a chance to (I think it was left over from an earlier iteration or something)

**Pull request merged**